provider/baked/src/zerotrie.rs

```diff
@@ -77,6 +99,7 @@ impl<M: DataMarker> super::DataStore<M> for Data<M> {
         } else {
             cursor.take_value()
         }
+        // Safety: Allowed since `i` came from the trie and the field safety invariant
         .map(|i| unsafe { self.values.get_unchecked(i) })
```

Comment:
thought: maybe this unchecked is not worth the trouble

Reply:
I'm happy to remove it if you are, but this is in a hot path I think